### ISSUE TITLE
Crash on exit when application is closed shortly after startup

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#2783]
**[Submitted to the original trac issue database at 11.35am, Tuesday, 9th March 2010]**

Version: svn 20382

How to repeat:

Once the application has started and the initial window has appeared, hit the close-window icon.  I think you need to close the window within 3 seconds of appearing to trigger this.

```
(gdb) bt
#0  0xb7fa0410 in __kernel_vsyscall ()
#1  0xb560d085 in raise () from /lib/tls/i686/cmov/libc.so.6
#2  0xb560ea01 in abort () from /lib/tls/i686/cmov/libc.so.6
#3  0xb5645b7c in ?? () from /lib/tls/i686/cmov/libc.so.6
#4  0xb564da85 in ?? () from /lib/tls/i686/cmov/libc.so.6
#5  0xb56514f0 in free () from /lib/tls/i686/cmov/libc.so.6
#6  0xb5649f43 in _IO_free_backup_area () from /lib/tls/i686/cmov/libc.so.6
#7  0xb5647ed9 in _IO_file_overflow () from /lib/tls/i686/cmov/libc.so.6
#8  0xb564740e in _IO_file_xsputn () from /lib/tls/i686/cmov/libc.so.6
#9  0xb5623ae3 in vfprintf () from /lib/tls/i686/cmov/libc.so.6
#10 0xb5629322 in fprintf () from /lib/tls/i686/cmov/libc.so.6
#11 0x08134763 in myMessageOutput (msgType=QtDebugMsg, buf=0x863ee80 "Reindexing... ") at Main.cpp:40
#12 0xb58d38a5 in qt_message_output (msgType=QtDebugMsg, buf=0x863ee80 "Reindexing... ") at global/qglobal.cpp:2196
#13 0x08075aa6 in ~QDebug (this=0xbfc9878c) at /usr/local/Trolltech/Qt-4.6.2/include/QtCore/qdebug.h:85
#14 0x0820fdb2 in Layer::reIndex (this=0x84fd310) at Layers/Layer.cpp:410
#15 0x082130b4 in Layer::clear (this=0x84fd310) at Layers/Layer.cpp:261
#16 0x0822a642 in ~ImageMapLayer (this=0x84fd310) at Layers/ImageMapLayer.cpp:70
#17 0x08075cb3 in ~MapDocumentPrivate (this=0x884ffe0) at Document.cpp:41
#18 0x08070ad9 in ~Document (this=0x8795840) at Document.cpp:77
#19 0x08148ac6 in ~MainWindow (this=0xbfc98904) at MainWindow.cpp:360
#20 0x08134570 in main (argc=Cannot access memory at address 0x4715
) at Main.cpp:228
```

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 1.41pm, Tuesday, 9th March 2010]**

(In [20386]) FIX : Do not count deleted parents when removing children nodes (fixes #2785)
FIX : Make sure not to write to the log after it has been closed (fixes #2783)

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 1.45pm, Tuesday, 9th March 2010]**

(In [20387]) FIX : Do not count deleted parents when removing children nodes (fixes #2785)
FIX : Make sure not to write to the log after it has been closed (fixes #2783)
